frontend/src/core/api/types.js

```diff
@@ -56,24 +56,8 @@ export type OtherLocaleTranslation = {|
  * Translation Memory... ).
  */
 export type MachineryTranslation = {|
-    sources: Array<{
-    type: 'translation-memory',
-        itemCount: number,
-    }|{
-        type: 'google-translate',
-    }|{
-        type: 'microsoft-translator',
-    }|{
-        type: 'microsoft-terminology',
-        sourceString: string,
-        localeCode: string,
-    }|{
-    type: 'transvision',
-        sourceString: string,
-        localeCode: string,
-    }|{
-        type: 'caighdean'
-    }>,
+    sources: Array<{ type: 'translation-memory' | 'google-translate' | 'microsoft-translator' | 'microsoft-terminology' | 'transvision' | 'caighdean' }>,
```

Comment:
You can make this a bit simpler but making it an array of strings instead of an array of objects.

`sources: Array<'translation-memory' | 'google-translate' | 'microsoft-translator' | 'microsoft-terminology' | 'transvision' | 'caighdean'>,`

What would be even nicer would be to define a new type for the `sourceType`, and use that here instead of the long list of types.

Reply:
I tried to make it as an array of strings, but sources.type were needed somewhere else. I'll have a look at it and try to change it again.

Reply:
Oh it works with your suggestion `sources: Array<'translation-memory' | 'google-translate' | 'microsoft-translator' | 'microsoft-terminology' | 'transvision' | 'caighdean'>,`
Maybe I just mistyped something before.